client/src/RelatedItemsComparison/components/ProductCard/ProductCard.jsx

```diff
@@ -33,22 +37,25 @@ const StyledWrap = styled.div`
     transform: scale(1.2);
   }
 `;
+
 export default function ProductCard({
-  id, handleRemoveItemClick, symbol,
+  id, handleRemoveItemClick, symbol, handleStarClick,
 }) {
   const dispatch = useDispatch();
   const [photoURL, setPhotoURL] = useState('');
   const [category, setCategory] = useState('');
   const [name, setName] = useState('');
   const [price, setPrice] = useState(null);
   const [salePrice, setSalePrice] = useState('');
-  // const [avgRating, setAvgRating] = useState('');
-  const [showModal, setShowModal] = useState(false);
+  const [avgRating, setAvgRating] = useState('');
   const [cardClick, setCardClick] = useState(true);
 
-  // const calculateAvgRating = (ratings) => {
-  //   // calculate and return avg of all ratings in ratings obj
-  // };
+  const calculateAvgRating = (ratings) => {
+    const totalRatings = Object.values(ratings)
+      .reduce((a, b) => parseInt(a, 10) + parseInt(b, 10), 0);
+    const avg = ((ratings['1'] * 1) + (ratings['2'] * 2) + (ratings['3'] * 3) + (ratings['4'] * 4) + (ratings['5'] * 5)) / totalRatings;
```

Comment:
Be careful here, you're assuming ratings[value] will be a number. Sometime it is undefined. For example {1: 10, 3:10, 4: 10, 5:10}. Is this case, it'll end up as a string I believe.